File: js/src/2778-sumOfSquares.js
/**
 * @param {number[]} nums
 * @return {number}
 */
const sumOfSquares = function(nums) {
  const len = nums.length;
  let result = 0;
  for (let i = 0; i < len; i++) {
    if (len % (i + 1) === 0) {
      result += (nums[i] * nums[i]);
    }
  }
  return result;
};

export { sumOfSquares };

File: python/src/2778-sumOfSquares.py
class Solution:
    def sumOfSquares(self, nums: List[int]) -> int:
        result = 0
        for i in range(0, len(nums)):
            if (len(nums) % (i + 1) == 0):
                result += (nums[i] * nums[i])
        return result
